src/app/components/download/download.component.ts

```diff
@@ -0,0 +1,205 @@
+  public selectedSecondOrderField: ArlasSearchField;
+  public selectedThirdOrderField: ArlasSearchField;
+
+  public firstOrderColunm: number;
+  public secondOrderColunm: number;
+  public thirdOrderColunm: number;
+  public orderCommand: string;
+  public filterUrl: string;
+  public exportedTypeCommand: string;
+
+  public isCopied = false;
+
+  public server: any;
+
+  public exportTypeGroup: FormGroup;
+  public paramFormGroup: FormGroup;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private collaborativeService: ArlasCollaborativesearchService,
+    private configService: ArlasConfigService
+  ) { }
+
+  public ngOnInit() {
+    this.exportTypeGroup = this.formBuilder.group({
+      exportType: ['', Validators.required]
+    });
+    this.paramFormGroup = this.formBuilder.group({
+      availableFields: ['', Validators.required],
+      orderField: [''],
+      orderField2: [{ value: '', disabled: true }],
+      orderField3: [{ value: '', disabled: true }]
+    });
+
+    this.server = this.configService.getValue('arlas.server');
+    this.collaborativeService.describe(this.server.collection.name).subscribe(
+      description => {
+        const fields = description.properties;
+        Object.keys(fields).forEach(fieldName => {
+          this.getFieldProperties(fields, fieldName);
+        });
+      },
+      error => {
+        this.collaborativeService.collaborationErrorBus.next(error);
+      });
+  }
+
+  public onSelectionChange(selectedOptionsList) {
+    this.selectedFields = new Array<ArlasSearchField>();
+    this.selectedFieldString = '';
+    selectedOptionsList.forEach((option, index) => {
+      const field = option._element.nativeElement.textContent.split('-');
+      this.selectedFields.push(new ArlasSearchField(field[0].trim(), field[1].trim()));
+      this.selectedFieldString += (index !== 0 ? ',' : '') + field[0].trim();
+    });
+    this.selectedFirstOrderField = null;
+    this.selectedSecondOrderField = null;
+    this.selectedThirdOrderField = null;
+  }
+
+  public orderSelect() {
+    if (this.selectedFirstOrderField) {
+      this.paramFormGroup.get('orderField2').enable();
+    } else {
+      this.paramFormGroup.get('orderField2').disable();
+      this.selectedSecondOrderField = null;
+    }
+
+    if (this.selectedSecondOrderField) {
+      this.paramFormGroup.get('orderField3').enable();
+    } else {
+      this.paramFormGroup.get('orderField3').disable();
+      this.selectedThirdOrderField = null;
+    }
+  }
+
+  /**
+   * Switches between dialog steps
+   * @param event The step index
+   */
+  public changeStep(event) {
+    if (event.selectedIndex === 2) {
+      this.isCopied = false;
+      this.exportedTypeCommand = this.exportTypeGroup.get('exportType').value;
+      this.orderCommand = '';
+      const filters = new Array<Filter>();
+      this.collaborativeService.collaborations.forEach(element =>
+        filters.push(element.filter)
```

Comment:
Maybe we can use .map function to create filters array ?
const filters = this. collaborativeService.collaborations.map(element=>element.filter)

Reply:
`this. collaborativeService.collaborations` is a Map, map() operator is not applicable here